real_check/real_check.ps1

```diff
@@ -0,0 +1,57 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+function Check-Symbols {
+    param(
+        [string] $lib
+    )   
+        try {
+            [string]$output = cmd.exe /c dumpbin /ALL $lib
+        }
+        catch {
+            Write-Error  "Unable to run dumpbin on given lib"
+            return 2
+        }
+        $start_token = "public symbols"
+        $end_token = "Archive member name"
+        try {
+            [int] $start = $output.IndexOf($start_token)
+            [int] $end = $output.IndexOf($end_token, $start)
+            $symbol_text = ($output.Substring($start + $start_token.Length+1, $end-$start-$start_token.Length-1))
+            $split_symbols = $symbol_text.Split('',[StringSplitOptions]::RemoveEmptyEntries)
+        }
+        catch {
+            Write-Error  "Unable to parse dumpbin output"
+            return 2
+        }
+        if($split_symbols.Length -lt 2) {
+            Write-Error  "Unable to parse dumpbin output"
+            return 2
+        }
+        
+        $real_prefix = "real_"
+        $symbol_table = New-Object -TypeName "System.Collections.Generic.HashSet[string]"
+        $duplicate_symbols = New-Object -TypeName "System.Collections.ArrayList"
+    
+        $return_code = 0
+        for($i=0; $i -lt $split_symbols.Length; $i++){
+            if($i % 2 -ne 0) {
+                if($split_symbols[$i].StartsWith($real_prefix)) {
+                    $real_symbol = $split_symbols[$i]
+                    $symbol = $split_symbols[$i].Substring($real_prefix.Length)
+                } else {
+                    $symbol = $split_symbols[$i]
+                    $real_symbol = $real_prefix + $symbol
+                }
+                if(-not ($symbol_table.Add($symbol) -and $symbol_table.Add($real_symbol))){
+                    $return_code = 1
+                    [void]$duplicate_symbols.Add($symbol)
```

Comment:
>$duplicate_symbols [](http://example.com/codeflow?start=26&length=18)

To be fair duplicate symbols in the linker land means somthing else usually. It means that 2 object files have the same symbol and at link time the linker finds a duplicate symbol.

But giventhe fact that we can print directly here the variable $duplicate_symbols is not needed so it can be removed.

Reply:
And I would print something like:
"Original symbol and real symbol found: "

Reply:
Updated the error message.